include/ck/tensor_operation/gpu/device/impl/device_softmax_impl.hpp

```diff
@@ -257,40 +258,78 @@ struct DeviceSoftmaxImpl : public DeviceSoftmax<InDataType,
         };
     };
 
-    bool IsSupportedArgument(const BaseArgument* p_arg) override
+    static bool IsSupportedArgument(const Argument& arg)
     {
-        const Argument* p_arg_ = dynamic_cast<const Argument*>(p_arg);
-
         if constexpr(InSrcVectorDim == 0)
         {
-            if constexpr(NumInvariantDim == 0)
+            if constexpr(kNumInvariantDim == 0)
             {
                 return false;
             }
             else
             {
-                if(p_arg_->inStrides_[NumInvariantDim - 1] != 1)
+                if(arg.inStrides_[kNumInvariantDim - 1] != 1 && InSrcVectorSize != 1)
+                {
                     return false;
-
-                if(p_arg_->invariant_lowest_length_ % InSrcVectorSize != 0)
+                }
+                if(arg.invariant_lowest_length % InSrcVectorSize != 0)
+                {
                     return false;
-            };
+                }
+            }
         }
         else
         {
-            if(p_arg_->inStrides_[Rank - 1] != 1)
+            if(arg.inStrides_[Rank - 1] != 1 && InSrcVectorSize != 1)
+            {
                 return false;
-
-            if(p_arg_->inLengths_[Rank - 1] % InSrcVectorSize != 0)
+            }
+            if(arg.reduce_lowest_length % InSrcVectorSize != 0)
```

Comment:
`reduce_lowest_length` is not defined, which causes compiling error.   Not completely consistent synchronization with the latest develop branch

Reply:
Keep use `p_arg_->inLengths_[Rank - 1]`

Reply:
Oh, I resolved merge conflicts too fast :) Fixed [here](https://github.com/ROCmSoftwarePlatform/composable_kernel/pull/406/commits/e39b065b550c34ca98cd3c927dffa69ff43aceab)